internal/gatewayapi/filters.go

```diff
@@ -839,22 +842,26 @@ func (t *Translator) processRequestMirrorFilter(
 	// Only add missing mirror destinations
```

Comment:
imo this entire logic of adding missing mirror destinations can be removed, and so you can simplify this logic

Reply:
> imo this entire logic of adding missing mirror destinations can be removed, and so you can simplify this logic

Are we building filterContext.Mirror from scratch and there is no duplicate Endpoints?
I definitely can refactor the code based on your assessment.

Reply:
yes we dont need to consider the duplicate case, please go ahead and refactor